### DISCUSSION TITLE
PItch Tuning

### DISCUSSION THREAD

**Discussion by Bumblebee001** [GrandOrgue/GoOdf#54] [General]
I tried applying pitch tuning (-12.000000) to a stop hoping to get 16' from 8'. HN set to 4. Nothing seems to happen! Have I forgotten something or is this a bug?

**Comment by larspalo** [comment]
Pitch tuning is in cent. To change an octave you need -1200. Don't change the harmonic number unless you actually move the samples to the new octave instead (ie. if you're loading the 048 sample as 036, which of course is perfectly possible to do).

**Comment by Bumblebee001** [comment]
Using that same logic I assume that if I create another stop from octave 4
I set the HN to 16 even though I change pitch tuning.

**Comment by larspalo** [comment]
The harmonic number (together with either embedded pitch info or odf supplied pitch info, _and_ possibly pitch correction) are used when re-tuning in temperaments (other than original). If the stop/pipe is intending to be at corresponding 4' pitch (relative to 036-C) then HN should be 16, yes. But if you have the samples for a 4' stop and then load them starting at 048 (or pipe 13 and with its pipe sample number still as 036, which normally would have been pipe 1) then the 4' samples suddenly become an 8' stop and should have a HN of 8.

Pitch tuning is used for adjusting pitch in the "original" temperament.

**Reply by Bumblebee001** [reply]
What I'm trying to do is use the samples as they are as the source and retuning them all at -1200. So was I correct to adjust the HN accordingly? All the pipes are as they are from 036 upwards. and each will be returned by applying the appropriate pitch correction.

Another question arises: how best to extend 56 samples to 61 short of creating the missing notes using an audio editor?

**Reply by larspalo** [reply]
> So was I correct to adjust the HN accordingly?

Yes, if the whole "new" stop is of that pitch the HN should reflect it. What you intend to do could be done simply by loading the pipes starting with one octave "shift", and then only extrapolate the missing notes instead of pitch adjusting the whole "rank". (the pitch adjusting is more likely to introduce aliasing than if you load the samples "shifted")

> Another question arises: how best to extend 56 samples to 61 short of creating the missing notes using an audio editor?

The best option for audio quality is to create all the notes pitch shifted (or extrapolated) as you suggest in an audio editor.

Frequently it's deemed enough to just "transpose" the top 5 samples so that 087-D# is loaded again as 092 with PitchTuning=500 and so on up to 091-G that's loaded as 096 (also with PitchTuning=500). Provided the pitch info is correct in the samples no further adjustment is needed for temperament, but the actual pitch tuning transposing of 500 might need some fine adjustment to be exactly in pitch.

When I did the Norrfjärden extension with almost an extra octave I did create the 087-D#, 088-E, 093-A and 094-Bb in an audio editor and then used each of the samples with PitchTuning=-200, as is, and finally with PitchTuning=200 (actual numbers may vary because of the meantone temperament). This strategy "should" introduce less aliasing than pitch shifting an octave and still less extra samples than if each was created. Also my intention was to keep more of the C/C# pipe layout in the stereo image.

**Comment by Bumblebee001** [comment]
Thanks for the explanation. If one had to use your software to do the job,
is it possible? Can one reload the last few notes for the missing ones and
adjust accordingly?

**Reply by larspalo** [reply]
It's possible to do, but at the moment it requires manual adding of the pipes, which is safe but tedious. I have planned a more flexible copying and loading option that could handle such things, but at the moment the fastest strategy is unfortunately to use a text editor and adjust the values manually...

The way to currently add 5 more top notes in a stop from within GOODF is: 

- First adjust the number of pipes from 56 to 61
- Select pipe 57
- Hit Ctrl+A and select the attack to add
- If separate releases should be added it's done by selecting the pipe and hitting Ctrl+R and selecting release
- Adjust pipe properties with selected pipe and hitting Ctrl+E
- Repeat steps as necessary for pipe 58, 59, 60 and 61

In the future, as discussed in https://github.com/larspalo/GOODF/discussions/51#discussioncomment-6708658 I want a more flexible loading and copying option.

**Comment by Bumblebee001** [comment]
Great! That in itself helps to understand the program better in case one needs to make some other specific adjustment to a specific note. One must bear in mind that when creating new notes from an actual recording the loops in particular have to be redone. What was correct in one, changes completely upon tuning up or down and clicks appear. At least that has been my experience. I have also experienced complete disappearance of loops and markers after re-processing a note in an audio editor. So one is better off starting from a clean sample, retune and create markers, cut out and treat the releases, select loops, save correct pitch information basically
treating each new note as though a new recording. 

That’s a lot of work compared to creating a few lines in an odf.

Mark

**Comment by rsporken** [comment]
I opened the ODF for St John Cantius Krakov (P. Grabowski) in GOODF. After saving the file, some notes in the lower pedal octave are wrong. Then I noticed in ODFEdit that some pipes had a pitch tuning of more than 100 cents. Looking at the same file in GOODF shows pitch tuning of 0 cents for all pipes. Does anyone know 1) why some pipes are detuned by 1/2 or 1 tone? 2) why this information seems to be lost in GOODF? Thanks?

**Comment by larspalo** [comment]
@rsporken Usually, when the PitchTuning value is about 100 cent or more it's because the sample is "transposed" to another tone (pitch) as a way to re-use the same sample for multiple tones that perhaps didn't exist in the original organ or where the original sample needed to be exchanged for some reason.

Are you _really_ sure that the PitchTuning values existed in the same .organ file that you open in GOODF and that they have disappeared by themselves when you saved the file? I cannot reproduce such an issue here so I'd like to know more about it and preferably get an example file that will exhibit the problem when I open it in GOODF and for instance save it back under another name. Could you create a minimal .organ file (without any real samples except perhaps empty .wav files) that behaves in the same way and post it to me so that I can examine it?

**Reply by larspalo** [reply]
@rsporken I'm thinking of something like this, but showing the problem you're experiencing. Try opening the .organ file in GOODF and change the name of the .organ file and save it back and compare the PitchTuning lines of the original file to the newly saved one. For me they are identical.

[TestPT.zip](https://github.com/larspalo/GOODF/files/12884458/TestPT.zip)

**Reply by rsporken** [reply]
Actually, I think I was wrong in my first analysis. GOODF does not change the pitch tuning values. I must have done something else that changed them, but I have not yet been able to reproduce the problem. However, I cannot see the pitch tuning values in GOODF; that made me think it actually changed the values. When I open your example in GrandOrgue or when I look at it in ODFEdit, I can see the pitch tuning. With GOODF, they all show as 0 (see attached files). Maybe I am not looking in the right place.
Thanks for your help.
![Example GOODF](https://github.com/larspalo/GOODF/assets/147702424/863f5345-93d7-4b9d-9844-26ce004ac9ea)
![Example ODFEdit](https://github.com/larspalo/GOODF/assets/147702424/85053198-a325-47e4-83b9-ca967c617c5c)

**Reply by larspalo** [reply]
@rsporken Yes, you're looking at the rank (internal rank of Stop) level that's displayed above the pipe tree and pipe loading options. To see the pitch tuning for the individually selected pipe, you must open the pipe properties dialog. (Select a pipe and then hit Ctrl+E, or right click the pipe and select "Edit pipe properties" in the pop-up menu).

If you find anything problematic that you can re-produce, feel free to open an issue about it.

**Reply by rsporken** [reply]
Great, now I know how to do that. Thanks very much again!
:)

**Comment by larspalo** [comment]
With 82a92d958a7fdbd12f1c9ec76bd862716ca01437 it's now much easier to for instance extend the compass of an organ. This won't be available in a build until a new release is made though, if one doesn't build from current source. The sequence to extend from 56 notes to 61 will be:

- Adjust number of logical pipes to 61
- Select the pipe to copy from in the pipe tree
- Hit Ctrl+C _or_ right click the pipe and select "Copy this pipe offset to..."
- In the new dialog select the target pipe(s) and then push "Copy to selected" button

The PitchTuning will be automatically adjusted for each target pipe. This works best for equal temperament samples if any arbitrary source pipe is chosen, or else, it's best to transpose an octave so that the same tone name will be used.
